### PR TITLE
Added WPF/GTK/macOS nodes to all samples

### LINKED ISSUE #55: Add WPF, WinForms, Mac nodes to samples
From slack discussion with @charlesroddie and @mydogisbox 

> The pre-release of Xamarin.WPF is just out. It seems to be usable with glitches here or there as you would expect. It has all the stock controls in Xamarin.Forms. There is almost no support in 3rd party controls at the moment. (edited)

Let's add a sample under https://github.com/fsprojects/Elmish.XamarinForms/tree/master/Samples/CounterApp

I'd like to also see WinForms and Mac nodes under there.

Here's a sample from @mydogisbox:

```

type CounterApp () as app = 

    inherit Application ()
    let program = Program.mkProgram App.init App.update App.view
    let runner = 
        program
        |> Program.withConsoleTrace
        |> Program.withDynamicView app
        |> Program.run
 

type MainWindow() as this =
    inherit FormsApplicationPage()

    do Forms.Init()
    do this.LoadApplication(new CounterApp())


[<STAThread>] 
[<EntryPoint>]
do (new System.Windows.Application()).Run(MainWindow()) |> ignore
```

### PR DESCRIPTION
**Pull request by TimLariviere**
Closes #55 
I didn't add WinForms because it's still not available on NuGet.
https://github.com/aosoft/Xamarin.Forms.WinForms/issues/6

Also I've added missing tests for the macOS and GTK templates in the build script.
But these tests are currently failing, and I've trouble finding out why.

### CODE REVIEW

**Comment by TimLariviere** [comment]
Strange, The GTK template was missing `nuget.targets` which should do this exact thing, and I've added it.
Ok, I will try tonight with your suggestion.

The macOS template also fails. It either complains that there's no Build target or that the shared project dll is missing...

**Comment by mallibone** [comment]
I tried it on your branch yesterday, that one line fixed it for me... but perhaps I can squeeze in some 20 minutes today to give it a go from a clean checkout.
Whoever added that GTK template initially must have missed the `nuget.targets` line 😇

File: install-xamarin-android-ios.sh [hunk 1]
```diff
@@ -1,5 +1,6 @@
 XAMARIN_IOS_VERSION=11.10.1.178
 XAMARIN_ANDROID_VERSION=8.3.0-19
+XAMARIN_MAC_VERSION=4.8.0.22
```
Comment (mallibone) [inline install-xamarin-android-ios.sh]: I think you will have to take the lower version [4.5.0.280](https://docs.microsoft.com/en-us/xamarin/mac/release-notes/4/4.5) all above require XCode 9.4 or higher.

Comment (TimLariviere) [reply install-xamarin-android-ios.sh]: Ok. I changed it.

**Comment by TimLariviere** [comment]
Travis CI once again failed to download the SDKs.
```
--2019-01-23 14:08:22--  https://dl.xamarin.com/MonoTouch/Mac/xamarin.ios-11.10.1.178.pkg

Resolving dl.xamarin.com (dl.xamarin.com)... failed: nodename nor servname provided, or not known.

wget: unable to resolve host address â€˜dl.xamarin.comâ€™

--2019-01-23 14:08:22--  https://dl.xamarin.com/MonoforAndroid/Mac/xamarin.android-8.3.0-19.pkg

Resolving dl.xamarin.com (dl.xamarin.com)... failed: nodename nor servname provided, or not known.

wget: unable to resolve host address â€˜dl.xamarin.comâ€™

--2019-01-23 14:08:23--  https://dl.xamarin.com/XamarinforMac/Mac/xamarin.mac-4.8.0.22.pkg

Resolving dl.xamarin.com (dl.xamarin.com)... failed: nodename nor servname provided, or not known.

wget: unable to resolve host address â€˜dl.xamarin.comâ€™
```


#285 is deeply needed :)

**Pull request merged**